### ISSUE TITLE
Optimize iteration over the keys in the database

### ISSUE THREAD

**Issue by xgreenx** [FuelLabs/fuel-core#1997]
We have many cases when we need to iterate only over keys of the specific table/column. The current iteration logic loads both key and values into memory and later discards values. To avoid vesting of resources, we can add more functionality and provide iteration only over the keys.

As an example:
When we want to get the last block height, we load the whole `CompressedBlock` into the memory, when we don't need it.

**Comment by vishnuc77** [comment]
Hi @xgreenx, I'm a first time contributor trying to get involved with the project.
I would like to know whether this task is something like making use of `iter_all_keys()` in place of (and other similar places) :
```
pub fn latest_height(&self) -> StorageResult<BlockHeight> {
        self.iter_all::<FuelBlocks>(Some(IterDirection::Reverse))
            .next()
            .ok_or(not_found!("BlockHeight"))?
            .map(|(height, _)| height)
 }
 ```
 so that CompressedBlock itself would not be loaded into memory, or is there something more to it?

**Comment by rymnc** [comment]
taking this up 😁

**Comment by rymnc** [comment]
should we mark this as closed with #2092?
cc: @xgreenx

**Comment by xgreenx** [comment]
Closed via https://github.com/FuelLabs/fuel-core/pull/2092
